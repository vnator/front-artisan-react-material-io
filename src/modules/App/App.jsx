import React from 'react';
import {
  createMuiTheme,
  ThemeProvider,
  responsiveFontSizes,
  makeStyles,
} from '@material-ui/core/styles';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { useIntl } from 'react-intl';

import style from './App.module.css';
import { theme } from '../../theme';
import { Counter } from '../../components/Counter/Counter';
import { Address } from '../../pages/Address';
import { Header } from '../Header/Header';
import { MAIN_ROUTES } from '../../const/routes';
import { UserList } from '../../pages/UserList/UserList';
import { User } from '../../pages/User';
import { Toast } from '../../components/Toast/Toast';
import { Nav } from '../Nav/Nav';
import { Container, Typography, Link, Paper } from '@material-ui/core';

const useStyles = makeStyles(props => ({
  app: {
    textAlign: 'center',
    background: props.palette.grey['200'],
    height: '100vh',
  },
  message: {
    padding: '30px 0',
  },
}));

const App = () => {
  const { formatMessage } = useIntl();
  const customTheme = responsiveFontSizes(createMuiTheme(theme));
  const classes = useStyles();

  return (
    <ThemeProvider theme={customTheme}>
      <div className={classes.app}>
        <Header />
        <Container>
          <Paper elevation={4}>
            <Typography
              className={classes.message}
              color="textPrimary"
              variant="body1">
              {formatMessage(
                {
                  id: 'app.paragraph',
                },
                {
                  extern: str => (
                    <Link
                      key={str}
                      className={style.link}
                      href="https://reactjs.org"
                      target="_blank"
                      rel="noopener noreferrer">
                      {str}
                    </Link>
                  ),
                  code: str => (
                    <code key={str} className={style.code}>
                      {str}
                    </code>
                  ),
                },
              )}
            </Typography>
            <Router>
              <Nav />
              <Switch>
                <Route exact path={MAIN_ROUTES.MAIN}>
                  <Counter />
                </Route>
                <Route path={MAIN_ROUTES.ADDRESS}>
                  <Address />
                </Route>
                <Route path={MAIN_ROUTES.USER_LIST}>
                  <UserList />
                </Route>
                <Route path={MAIN_ROUTES.USER()}>
                  <User />
                </Route>
              </Switch>
            </Router>
          </Paper>
          <Toast />
        </Container>
      </div>
    </ThemeProvider>
  );
};

export { App };
