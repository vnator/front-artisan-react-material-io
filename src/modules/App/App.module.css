.App {
  text-align: center;
  min-height: 100vh;
  background-color: var(--first-dark);
}

.paragraph {
  color: var(--white-full);
  margin: 0;
}

.code {
  font-weight: var(--weight-medium);
}

.link {
  color: var(--second);
}

.nav {  
  display: flex;
  align-items: center;
  justify-content: center;
  list-style: none;
  padding: 0;
}

.opt {
  width: 120px;
  margin: var(--space-large) 0;
}

.opt a {
  display: block;
  text-transform: uppercase;
  color: white;
  font-weight: bold;
  text-align: center;
  transition: var(--transition-base);
}

.opt a:hover {
  opacity: 0.4;
}
