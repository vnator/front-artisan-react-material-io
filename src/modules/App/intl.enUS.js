import { ERROR } from '../../const/errors';

const { TYPE, CODE } = ERROR;

const app_enUS = {
  paragraph:
    'Edite <code>src/modules/App/App.js</code> e salve para recarregar. <extern>Aprenda React</extern>',
  error: {
    [TYPE.INPUT]: {
      [CODE.A01]: `${TYPE.INPUT}-${CODE.A01}: Type a valid e-mail`,
    },
    [TYPE.AUTH]: {
      [CODE.A01]: `${TYPE.AUTH}-${CODE.A01}: User not authenticated`,
    },
    [TYPE.CRASH]: {
      [CODE.A01]: `${TYPE.CRASH}-${CODE.A01}: Invalid street name`,
    },
    [TYPE.FETCH]: {
      [CODE.A01]: `${TYPE.FETCH}-${CODE.A01}: User with id: {user} don't found`,
      [CODE.A02]: `${TYPE.FETCH}-${CODE.A02}: User with e-mail: {user} don't found`,
      [CODE.A03]: `${TYPE.FETCH}-${CODE.A03}: User {user} don't registred`,
    },
  },
};

export { app_enUS };
