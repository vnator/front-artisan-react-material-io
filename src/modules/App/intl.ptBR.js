import { ERROR } from '../../const/errors';

const { TYPE, CODE } = ERROR;

const app_ptBR = {
  paragraph:
    'Edite <code>src/modules/App/App.js</code> e salve para recarregar. <extern>Aprenda React</extern>',
  menu: {
    counter: 'counter',
    address: 'address',
    users: 'users',
  },
  error: {
    [TYPE.INPUT]: {
      [CODE.A01]: `${TYPE.INPUT}-${CODE.A01}: Type a valid e-mail`,
    },
    [TYPE.AUTH]: {
      [CODE.A01]: `${TYPE.AUTH}-${CODE.A01}: User not authenticated`,
    },
    [TYPE.CRASH]: {
      [CODE.A01]: `${TYPE.CRASH}-${CODE.A01}: Invalid street name`,
    },
    [TYPE.FETCH]: {
      [CODE.A01]: `${TYPE.FETCH}-${CODE.A02}: User with {user} don't found`,
      [CODE.A02]: `${TYPE.FETCH}-${CODE.A02}: User with {user} don't found`,
      [CODE.A03]: `${TYPE.FETCH}-${CODE.A03}: User {user} don't registred`,
    },
  },
};

export { app_ptBR };
