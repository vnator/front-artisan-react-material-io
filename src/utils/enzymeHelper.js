import { IntlProvider, createIntl } from 'react-intl';
import { mount, shallow } from 'enzyme';
import { flattenMessages } from '../config/flattenMessages';
import { messages } from '../messages';

// You can pass your messages to the IntlProvider. Optional: remove if unneeded.

const defaultLocale = 'pt-BR';
const locale = defaultLocale;
const wrapMessages = flattenMessages(messages[locale]);

function mountWithIntl(node) {
  return mount(node, {
    wrappingComponent: IntlProvider,
    wrappingComponentProps: {
      locale,
      defaultLocale,
      messages: wrapMessages,
    },
  });
}

function shallowWithIntl(node) {
  return shallow(node, {
    wrappingComponent: IntlProvider,
    wrappingComponentProps: {
      locale,
      defaultLocale,
      messages: wrapMessages,
    },
  });
}

const intl = createIntl({
  locale: 'pt-BR',
  defaultLocale: 'pt-BR',
  messages: wrapMessages,
});

export { mountWithIntl, shallowWithIntl, intl };
