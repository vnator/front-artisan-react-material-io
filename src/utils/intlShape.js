import Type from 'prop-types';

const intlConfig = {
  locale: Type.string,
  timeZone: Type.string,
  messages: Type.objectOf(Type.string),
  defaultLocale: Type.string,
};

const intlFormatters = {
  formatDate: Type.func,
  formatTime: Type.func,
  formatRelativeTime: Type.func,
  formatNumber: Type.func,
  formatPlural: Type.func,
  formatMessage: Type.func,
  formatHTMLMessage: Type.func,
};

const intlShape = {
  ...intlConfig,
  ...intlFormatters,
};

export { intlShape, intlConfig, intlFormatters };
