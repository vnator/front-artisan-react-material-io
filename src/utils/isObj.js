const isObj = obj => typeof obj === 'object' && !(obj instanceof Array);

export { isObj };
