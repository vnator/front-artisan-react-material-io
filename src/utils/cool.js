import { isObj } from './isObj';

const cool = (a) =>
  Object.freeze(
    Object.entries(a).reduce(
      (total, [key, val]) =>
        isObj(val)
          ? Object.freeze({ ...total, [key]: cool(val) })
          : Object.freeze({ ...total, [key]: val, isFrozen: true }),
      Object.freeze(),
    ),
  );

export { cool };
