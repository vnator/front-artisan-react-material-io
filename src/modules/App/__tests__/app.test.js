import React from 'react';
import { Provider } from 'react-redux';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { App } from '../App';
import { mountWithIntl } from '../../../utils/enzymeHelper';

Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureMockStore([thunk]);

describe('App', () => {
  it('Snapshot', () => {
    const store = mockStore({
      toast: { active: false, message: '' },
    });

    const app = mountWithIntl(
      <Provider store={store}>
        <App />
      </Provider>,
    );

    expect(app).toMatchSnapshot();
  });
});
