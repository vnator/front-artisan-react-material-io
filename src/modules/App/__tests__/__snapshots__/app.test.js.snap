// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot 1`] = `ReactWrapper {}`;
